### PR TITLE
ddpm custom timesteps

### PR DESCRIPTION
**Pull request by williamberman**
Many diffusion codebases allow configuring timesteps with varying spacing. This is a snippet that I have been copy and pasting in for testing purposes. 

Our standard method of calculating the previous timestep is using the constant step size. When using custom timesteps, we have to search in the tensor for the previous index.

### CODE REVIEW

**Review by sayakpaul — COMMENTED** [review]
LGTM desing-wise.

Tests would be helpful for folks willing see how to use custom step spacing.

File: src/diffusers/schedulers/scheduling_ddpm.py [hunk 1]
```diff
@@ -181,14 +182,31 @@ def scale_model_input(self, sample: torch.FloatTensor, timestep: Optional[int] =
         """
         return sample
 
-    def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.device] = None):
+    def set_timesteps(
+        self,
+        num_inference_steps: int,
+        device: Union[str, torch.device] = None,
+        custom_timesteps: Optional[List[int]] = None,
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddpm.py:189]:
```suggestion
        timesteps: Optional[List[int]] = None,
```
I don't think we need to call it `custom_...` given that the method is called `set_timesteps(...)` 

Also let's make sure that `num_inference_steps` and `timesteps` cannot both be set -> in this case we should throw an error. We should also update the docstring here stating that either `num_inference_steps` or `timesteps` has to be set and explain the difference.

File: src/diffusers/schedulers/scheduling_ddpm.py [hunk 2]
```diff
@@ -181,14 +182,31 @@ def scale_model_input(self, sample: torch.FloatTensor, timestep: Optional[int] =
         """
         return sample
 
-    def set_timesteps(self, num_inference_steps: int, device: Union[str, torch.device] = None):
+    def set_timesteps(
+        self,
+        num_inference_steps: int,
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddpm.py:187]:
```suggestion
        num_inference_steps: Optional[int] = None,
```

File: src/diffusers/schedulers/scheduling_ddpm.py (hunk 1 above)
Comment (williamberman) [reply src/diffusers/schedulers/scheduling_ddpm.py:189]: Yep, agreed!

File: src/diffusers/schedulers/scheduling_ddpm.py (hunk 2 above)
Comment (williamberman) [reply src/diffusers/schedulers/scheduling_ddpm.py]: done!

**Review by patrickvonplaten — APPROVED** [review]
Looks nice!

**Pull request merged**